mlir/lib/Dialect/SCF/IR/SCF.cpp

```diff
@@ -1787,9 +1787,9 @@ struct ForallOpReplaceConstantInductionVar : public OpRewritePattern<ForallOp> {
       }
       rewriter.replaceAllUsesWith(
           iv, getValueOrCreateConstantIndexOp(rewriter, loc, lb));
-      return success();
+      changed = true;
     }
-    return failure();
+    return success(changed);
   }
```

Comment:
In this commit you removed an early return: can you add a test demonstrating the behavior change here?

Reply:
There wouldn't be a failure before the change. The change just makes the pattern run fewer times to achieve the same result.

Reply:
So this patch is NFC? It would help in the future if you tag the PR as such when there is no behavior change :)